src/bridge/bridge.c

```diff
@@ -174,145 +174,71 @@ send_init_command (CockpitTransport *transport,
-  g_autofree gchar *address = g_data_input_stream_read_line (dbus_output, NULL, NULL, &error);
-
-  if (address)
-    {
-      cockpit_setenv_check ("DBUS_SESSION_BUS_ADDRESS", address, TRUE);
-      g_debug ("session bus address: %s", address);
-    }
-  else
-    {
-      if (error)
-        g_warning ("couldn't read address from dbus-daemon: %s", error->message);
-      else
-        g_message ("dbus-daemon didn't send us a dbus address; not installed?");
-    }
-
-  return process;
-}
-
-static void
-setup_ssh_agent (gpointer data)
-{
-  g_unsetenv ("G_DEBUG");
-  prctl (PR_SET_PDEATHSIG, SIGTERM);
-  if (cockpit_close_range (3, INT_MAX, 0) < 0)
-    {
-      g_printerr ("couldn't close file descriptors: %m\n");
-      _exit (127);
-    }
-}
-
-static GPid
-start_ssh_agent (void)
-{
-  GError *error = NULL;
-  GPid pid = 0;
-  gint fd = -1;
-  gint status = -1;
-
-  gchar *pid_line = NULL;
-  gchar *agent_output = NULL;
-  gchar *agent_error = NULL;
-  gchar *bind_address = g_strdup_printf ("%s/ssh-agent.XXXXXX", g_get_user_runtime_dir ());
-
-  gchar *agent_argv[] = {
-      "ssh-agent",
-      "-a",
-      bind_address,
-      NULL
-  };
+  g_debug ("launched %s: %s", argv[0], g_subprocess_get_identifier (process));
 
-  fd = g_mkstemp (bind_address);
-  if (fd < 0)
-    {
-      g_warning ("couldn't create temporary socket file: %s", g_strerror (errno));
-      goto out;
-    }
-  if (g_unlink (bind_address) < 0)
+  /* get the first line of output to figure out the socket address */
+  g_autoptr (GDataInputStream) stream = g_data_input_stream_new (g_subprocess_get_stdout_pipe (process));
+  g_autofree gchar *first_line = g_data_input_stream_read_line (stream, NULL, NULL, &error);
+  if (!first_line)
     {
-      g_warning ("couldn't remove temporary socket file: %s", g_strerror (errno));
-      goto out;
+      g_warning ("couldn't read address from %s: %s", argv[0], error->message);
+      return NULL;
     }
 
-  if (!g_spawn_sync (NULL, agent_argv, NULL,
-                     G_SPAWN_SEARCH_PATH, setup_ssh_agent,
-                     NULL,
-                     &agent_output, &agent_error,
-                     &status, &error))
-    {
-      if (g_error_matches (error, G_SPAWN_ERROR, G_SPAWN_ERROR_NOENT))
-        g_debug ("couldn't start %s: %s", agent_argv[0], error->message);
-      else
-        g_warning ("couldn't start %s: %s", agent_argv[0], error->message);
-      goto out;
-    }
+  g_autoptr(GRegex) regex = g_regex_new (socket_pattern, G_REGEX_RAW, 0, &error);
+  g_assert_no_error (error);
 
-  if (!g_spawn_check_exit_status (status, &error))
+  g_autoptr(GMatchInfo) info = NULL;
+  if (!g_regex_match (regex, first_line, 0, &info))
     {
-      g_warning ("couldn't start %s: %s: %s", agent_argv[0],
-                 error->message, agent_error);
-      goto out;
+      g_warning ("output from %s didn't match expected pattern %s", argv[0], socket_pattern);
```

Comment:
This is certainly a Does Not Happen™ condition, but still -- would be nice to terminate and wait the process here.

Reply:
OK.  I did a force exit, and same for the similar case just above.